### DISCUSSION TITLE
撤回没有用

### DISCUSSION THREAD

**Discussion by Eysland** [mamoe/mirai#872] [Q&A] [answered]
（Java）希望收到指令后撤回那一条消息
使用`MessageSource.recall(event.getSource())`没有效果，并且我确定有权限撤回。
万分感谢
版本是2.0.0
日志的错误信息：
> [31m2021-01-17 23:42:13 E/Net 3613302284: Exception in NetworkHandler
> java.lang.NoSuchMethodError: 'void net.mamoe.mirai.message.data.MessageSource$Key.recall(net.mamoe.mirai.message.data.MessageSource)'
> 	at net.mamoe.mirai.message.data.MessageSource.recall(MessageSource.kt)
> 	at com.example.plugin.reply.lambda$onEnable$0(reply.java:86)
> 	at net.mamoe.mirai.event.EventChannel$subscribeAlways$2$invokeSuspend$$inlined$suspendCoroutine$lambda$1.run(EventChannel.kt:475)
> 	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:98)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:738)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
> 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
> [0m

**Comment by sandtechnology** [comment]
此问题应该被提交为issue，请提供完整网络日志和客户端截图 并附上使用的版本号

**Reply by Eysland** [reply]
更新了，不过客户端截图指？

**Reply by sandtechnology** [reply]
请确认运行时的mirai-core版本是否正确，我记得mcl下mirai-core是1.3.3的

**Reply by Eysland** [reply]
都是2.0.0，mcl的话，在libs文件夹里的也全都是2.0.0

**Reply by sandtechnology** [reply]
经测试确实无法使用MessageSource.recall(event.getSource())方法 @Him188 请求转为issue

**Comment by Him188** [comment, marked as answer]
是一个编译器 bug, https://github.com/Him188/kotlin-jvm-blocking-bridge/issues/7

**Reply by Him188** [reply]
会在 2.1.0 修复

**Reply by Eysland** [reply]
那，当前版本，有什么撤回的函数可以用

**Reply by Him188** [reply]
Mirai.getInstance().recallMessage

**Reply by sandtechnology** [reply]
当前版本中直接发送后的撤回是正常的，关于这个函数的曲线救国的话 我目前测试下来可以用recallin替代

**Reply by Eysland** [reply]
谢谢~
